infra/dev-personal/main.tf

```diff
@@ -128,3 +128,53 @@ module "psoxy-google-chat" {
     }
   }
 }
+
+locals {
+  # set this to no if it's the first time creating the slack function
+  slack_secret_created  = "no"
+  slack_service_account = "psoxy-slack-discovery-api@psoxy-dev-jose.iam.gserviceaccount.com"
+  slack_function_name   = "psoxy-slack-discovery-api"
+}
+
+# creates the secret, without versions. Only if local.slack_secret_created is set to "yes"
+module "slack-discovery-api-auth" {
+  source   = "../modules/gcp-oauth-long-access-strategy"
+
+  count = local.slack_secret_created == "yes" ? 0 : 1
+  project_id              = var.project_id
+  function_name           = local.slack_function_name
+  token_adder_user_emails = []
+}
+
+# reads the secret reference to bind it to slack connector cloud function
+module "slack-discovery-api-auth-read" {
```

Comment:
This is a read version of the module above. So it uses one or the other depending the local `slack_secret_created` to know if the resource is managed or not.

Reply:
Removed.